opentelemetry-api/src/opentelemetry/baggage/propagation/__init__.py

```diff
@@ -96,10 +106,17 @@ def fields(self) -> typing.Set[str]:
 
 
 def _format_baggage(baggage_entries: typing.Mapping[str, object]) -> str:
-    return ",".join(
-        key + "=" + urllib.parse.quote_plus(str(value))
-        for key, value in baggage_entries.items()
-    )
+
+    key_values = []
+
+    for key, value in baggage_entries.items():
+        # type: ignore
+        if _key_regex.match(key) is None or _value_regex.match(value) is None:
```

Comment:
Should this be done in `set_baggage` https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/src/opentelemetry/baggage/__init__.py#L58? A error/warning log would be nice when key/value are invalid. Also, I think we need to add a CHANGELOG entry for this fix.

Reply:
I think `set_baggage` is the correct place to validate. If we do that we can safely assume that baggage contents are valid and serialize them. The `extract` also uses the `set_baggage` and we would just return the same context when key/value are invalid. This requires no change in the propagator if handled in `set_baggage`.

Reply:
Also `set_baggage` could be called independently of `extract` if user wants to manually set baggage themselves in context.